### PR TITLE
Rail spec update

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

File: client/Wayland/wlfreerdp.c [hunk 1]
```diff
@@ -390,6 +391,22 @@ static BOOL handle_uwac_events(freerdp* instance, UwacDisplay* display)
...
+static BOOL handle_window_events(freerdp* instance)
+{
+	rdpSettings* settings;
+
+	if (!instance || !instance->settings)
+		return FALSE;
+
+	settings = instance->settings;
+
+	if (!settings->AsyncInput)
```
Comment (hardening) [inline client/Wayland/wlfreerdp.c]: AsyncInput is supposed to die soon..

File: channels/rail/client/rail_orders.c [hunk 2]
```diff
@@ -736,6 +754,155 @@ static UINT rail_recv_langbar_info_order(railPlugin* rail, RAIL_LANGBAR_INFO_ORD
...
+
+static UINT rail_recv_get_application_id_extended_response_order(railPlugin* rail,
+        RAIL_GET_APPID_RESP_EX* id,
+        wStream* s)
+{
+	RailClientContext* context = rail_get_client_interface(rail);
+	UINT error;
+
+	if (!context || !id)
+		return ERROR_INVALID_PARAMETER;
```
Comment (hardening) [inline channels/rail/client/rail_orders.c]: check on `id` is not needed

File: channels/rail/client/rail_orders.c [hunk 3]
```diff
@@ -736,6 +754,155 @@ static UINT rail_recv_langbar_info_order(railPlugin* rail, RAIL_LANGBAR_INFO_ORD
...
+	}
+
+	return error;
+}
+
+
+static UINT rail_read_get_application_id_extended_response_order(wStream* s,
+        RAIL_GET_APPID_RESP_EX* id)
+{
+	if (!s || !id)
```
Comment (hardening) [inline channels/rail/client/rail_orders.c]: check on `id`not needed

File: channels/rail/client/rail_orders.c [hunk 4]
```diff
@@ -736,6 +754,155 @@ static UINT rail_recv_langbar_info_order(railPlugin* rail, RAIL_LANGBAR_INFO_ORD
...
+}
+
+static UINT rail_recv_power_display_request_order(railPlugin* rail,
+        RAIL_POWER_DISPLAY_REQUEST* power,
+        wStream* s)
+{
+	RailClientContext* context = rail_get_client_interface(rail);
+	UINT error;
+
+	if (!context || !power)
```
Comment (hardening) [inline channels/rail/client/rail_orders.c]: check on `power`not needed

File: channels/rail/client/rail_orders.c [hunk 5]
```diff
@@ -811,6 +978,24 @@ UINT rail_order_recv(railPlugin* rail, wStream* s)
 				return rail_recv_langbar_info_order(rail, &langBarInfo, s);
 			}
 
+		case RDP_RAIL_ORDER_ZORDER_SYNC:
+			{
+				RAIL_ZORDER_ZYNC val;
```
Comment (mfleisz) [inline channels/rail/client/rail_orders.c]: Typo: `RAIL_ZORDER_ZYNC` should be `RAIL_ZORDER_SYNC`

File: include/freerdp/rail.h [hunk 6]
```diff
@@ -352,26 +354,69 @@ struct _RAIL_COMPARTMENT_INFO_ORDER
 };
 typedef struct _RAIL_COMPARTMENT_INFO_ORDER RAIL_COMPARTMENT_INFO_ORDER;
 
+struct _RAIL_ZORDER_ZYNC
```
Comment (mfleisz) [inline include/freerdp/rail.h]: Typo

File: channels/rail/client/rail_orders.c (hunk 5 above)
Comment (akallabeth) [reply channels/rail/client/rail_orders.c]: @mfleisz done

File: channels/rail/client/rail_orders.c [hunk 7]
```diff
@@ -536,9 +546,17 @@ static UINT rail_recv_handshake_ex_order(railPlugin* rail, RAIL_HANDSHAKE_EX_ORD
...
+	 * Client response is really a Handshake PDU */
+	error = context->ClientHandshake(context, &clientHandshake);
+
+	if (error != CHANNEL_RC_OK)
+		return error;
+
 	if (context->custom)
 	{
-		IFCALLRET(context->ClientHandshakeEx, error, context, handshakeEx);
+		IFCALLRET(context->ServerHandshakeEx, error, context, handshakeEx);
```
Comment (m4ntis) [inline channels/rail/client/rail_orders.c]: Going through this PR for an implementation of rail server support I'm working on, noticed that you implemented here a change that got merged in #5345

**Pull request merged**